Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -874,16 +898,20 @@ void addMissingProductBonds(const Bond &origB, RWMOL_SPTR product,
   CHECK_INVARIANT(prodBeginIdxs.size() == prodEndIdxs.size(),
                   "Different number of start-end points for product bonds.");
   for (unsigned i = 0; i < prodBeginIdxs.size(); i++) {
-    product->addBond(prodBeginIdxs.at(i), prodEndIdxs.at(i),
-                     origB.getBondType());
+    addBondToProduct(origB, *product, prodBeginIdxs.at(i), prodEndIdxs.at(i));
   }
 }
 
 void addMissingProductAtom(const Atom &reactAtom, unsigned reactNeighborIdx,
                            unsigned prodNeighborIdx, RWMOL_SPTR product,
                            const ROMol &reactant,
                            ReactantProductAtomMapping *mapping) {
-  auto *newAtom = new Atom(reactAtom);
```

Comment:
couldn't this be:

```
auto *newAtom = reactAtom.copy();
```

Reply:
> couldn't this be:
> 
> ```
> auto *newAtom = reactAtom.copy();
> ```

yeah, that would be exactly equivalent. Feels less explicit though.

Reply:
I'm happy either way, it's probably "faster" as the query atom copy can do a static cast.

Reply:
It does feel more future proof though but I can't imagine a new Atom subclass so...